lib/segment/src/segment/snapshot.rs

```diff
@@ -160,6 +160,23 @@ impl Segment {
             .into_iter()
             .map(|file| (file, FileVersion::from(self.version_tracker.get_payload())));
 
+        let payload_index_files = self
+            .payload_index
+            .borrow()
+            .immutable_files()
+            .into_iter()
+            .map(|(field, file)| {
+                let version = FileVersion::from(
+                    self.version_tracker
+                        .get_payload_index(&field)
```

Comment:
A field may have multiple indices, of which only some are updated. Do we want to consider that case as well?

For example, we have a numeric and null index on `"a"`.

We currently have: `{"a": 123}`

If we update it to `{"a": 456}`, only the numeric index files are changed. The null index remains the same.